sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -187,6 +225,43 @@ public String getEventHubName() {
             .flatMap(node -> node.getPartitionProperties(partitionId));
     }
 
+    /**
+     * Get the idempotent producer's publishing state of a partition.
+     * @param partitionId The partition id of the publishing state
+     * @return A mono that has the {@link PartitionPublishingProperties}.
+     * {@code null} if the partition doesn't have any state yet.
+     * @throws UnsupportedOperationException if this producer isn't an idempotent producer.
+     */
+    Mono<PartitionPublishingProperties> getPartitionPublishingProperties(String partitionId) {
+        PartitionPublishingState publishingState = getClientPartitionPublishingState(partitionId);
+        if (publishingState.isFromLink()) {
+            return Mono.just(publishingState.toPartitionPublishingProperties());
+        } else {
+            return withRetry(this.getSendLink(partitionId).flatMap(amqpSendLink ->
+                    Mono.just(this.getClientPartitionPublishingState(partitionId))),
+                retryOptions.getTryTimeout(), retryPolicy).map(
+                    PartitionPublishingState::toPartitionPublishingProperties);
+        }
+    }
+
+    /**
+     * Get the idempotent producer's publishing state of a partition.
+     * @param partitionId The partition id of the publishing state
+     * @return A mono that has the {@link PartitionPublishingState}.
+     * {@code null} if the partition doesn't have any state yet.
+     * @throws UnsupportedOperationException if this producer isn't an idempotent producer.
+     */
+    Mono<PartitionPublishingState> getPartitionPublishingState(String partitionId) {
+        PartitionPublishingState publishingState = getClientPartitionPublishingState(partitionId);
```

Comment:
All of this should be wrapped in `Mono.defer()`.

Reply:
Fixed